oauth2_provider/oauth2_backends.py

```diff
@@ -120,9 +125,14 @@ def create_authorization_response(self, request, scopes, credentials, allow):
 
             # add current user to credentials. this will be used by OAUTH2_VALIDATOR_CLASS
             credentials["user"] = request.user
+            request_uri, http_method, _, request_headers = self._extract_params(request)
 
             headers, body, status = self.server.create_authorization_response(
-                uri=credentials["redirect_uri"], scopes=scopes, credentials=credentials
+                uri=request_uri,
```

Comment:
Hmm, the [documentation](https://oauthlib.readthedocs.io/en/latest/oauth2/endpoints/authorization.html) for this seems to indicate the previous behavior is correct:

```Post Authorization Request
Generally, this is where you handle the submitted form. Rather than using validate_authorization_request we use create_authorization_response which in the case of Authorization Code Grant embed an authorization code in the client provided redirect uri.```

Reply:
Does it? The code sample below that line explains it more clearly:

```python
# Validate request
uri = 'https://example.com/post_authorize?client_id=foo
headers, body, http_method = {}, '', 'GET'

# Fetch the credentials saved in the pre authorization phase
from your_datastore import fetch_credentials
credentials = fetch_credentials()

# Fetch authorized scopes from the request
from your_framework import request
scopes = request.POST.get('scopes')

from oauthlib.oauth2 import FatalClientError, OAuth2Error
from your_framework import http_response
http_response(body, status=status, headers=headers)
try:
    headers, body, status = server.create_authorization_response(
        uri, http_method, body, headers, scopes, credentials)
```

The `uri` passed to `create_authorization_response` in the example is clearly not the `redirect_uri`, its the URI of the current request. , `oauthlib` doesn't use the `uri` as the `redirect_uri`, it either extracts it from the uri or from the passed in `credentials`. We pass it in the `credentials`, which sets `request.redirect_uri` as an attribute on the `oauthlib.Request`

https://github.com/oauthlib/oauthlib/blob/b69fa53fd836dc559aa7fcd78ce075bcbe361629/oauthlib/oauth2/rfc6749/endpoints/authorization.py#L89-L95

Checking the redirect uri uses this request.redirect_uri attribute, not the request URI.

https://github.com/oauthlib/oauthlib/blob/b69fa53fd836dc559aa7fcd78ce075bcbe361629/oauthlib/oauth2/rfc6749/grant_types/base.py#L228-L241

Here are examples of how other libraries using oauthlib are handling this, from [flask-oauthlib](https://github.com/lepture/flask-oauthlib/blob/d8369c8ae06150d37da2eba951ae9646235f940b/flask_oauthlib/provider/oauth2.py#L493-L496):

```python
        uri, http_method, body, headers = extract_params()
        try:
            ret = server.create_authorization_response(
                uri, http_method, body, headers, scopes, credentials)
```

and from [bottle-oauthlib](https://github.com/Refinitiv/bottle-oauthlib/blob/d696c8388a8c499cb307f865f9264db47873cf6d/bottle_oauthlib/oauth2.py#L249-L255):

```python
                uri, http_method, body, headers = extract_params(bottle.request)
                scope = bottle.request.params.get('scope', '').split(' ')

                try:
                    resp_headers, resp_body, resp_status = self._oauthlib.create_authorization_response(
                        uri, http_method=http_method, body=body, headers=headers, scopes=scope
                    )
``` 

and from [oauthlib's test cases](https://github.com/oauthlib/oauthlib/blob/b69fa53fd836dc559aa7fcd78ce075bcbe361629/tests/oauth2/rfc6749/endpoints/test_client_authentication.py#L84-L90):

```python
        # implicit grant
        auth_uri = 'http://example.com/path?client_id=abc&response_type=token'
        self.assertRaises(ValueError, self.mobile.create_authorization_response,
                auth_uri, scopes=['random'])


        self.validator.validate_client_id.side_effect = self.set_client_id
        h, _, s = self.mobile.create_authorization_response(auth_uri, scopes=['random'])
```

None of them are passing the `redirect_uri` as the `uri` argument of `create_authorization_response`, and I don't see any bit of code or documentation that says we should.